### PR TITLE
make course search query part of query params

### LINKED ISSUE #41: Make course search query part of query params
On course list page: https://online.codingblocks.com/courses

If you search for anything, the results are displayed below. It works all right. However, if you reload the page resets and you need to retype the search query to get the results again. We aim to eliminate this by keeping the searchQuery are part of query params so that

```
https://online.codingblocks.com/courses?q=android
```

should auto fill the search query with value `android` and i would see android results. 

## Requirements
- Visiting https://online.codingblocks.com/courses?q=android should show android results
- Searching for `c++` in search box, the query params should also be updated along with the search results.

### PR DESCRIPTION
**Pull request by gurrrung**
closes #41 

- [x] Visiting `online.codingblocks.com/courses?q=android` should show android results
- [x] Searching for c++ in search box, the query params should also be updated along with the search results.

### CODE REVIEW

**Comment by gurrrung** [comment]
review ping @abhishek97

File: online.codingblocks.com/pages/courses/index.vue [hunk 1]
```diff
@@ -119,6 +119,10 @@ export default {
       return process.client ? this.$jsonApiStore.sync(this.featuredTagsPayload).sort((a,b)=>a.order - b.order) : []
     }
   },
+  mounted:function() {
```
Comment (abhishek97) [inline online.codingblocks.com/pages/courses/index.vue:122]: use es6 syntax 
```
mounted () {

}
```

File: online.codingblocks.com/pages/courses/index.vue [hunk 2]
```diff
@@ -103,7 +103,7 @@ export default {
       // featuredTags: [], // using this server sides fails, see computed property
       offset: 0,
       limit: 9,
-      searchQuery: '',
+      searchQuery: this.$route.query.q != undefined ? this.$route.query.q : '',
```
Comment (abhishek97) [inline online.codingblocks.com/pages/courses/index.vue:106]: can be simplified to 
```
this.$route.query.q || ''
```

File: online.codingblocks.com/pages/courses/index.vue (hunk 1 above)
Comment (gurrrung) [reply online.codingblocks.com/pages/courses/index.vue]: done

File: online.codingblocks.com/pages/courses/index.vue (hunk 2 above)
Comment (gurrrung) [reply online.codingblocks.com/pages/courses/index.vue]: Amazing, done.

**Pull request merged**